server/datastore/mysql/operating_systems.go

```diff
@@ -0,0 +1,144 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/jmoiron/sqlx"
+)
+
+func (ds *Datastore) ListOperatingSystems(ctx context.Context) ([]fleet.OperatingSystem, error) {
+	return listOperatingSystemsDB(ctx, ds.reader)
+}
+
+func listOperatingSystemsDB(ctx context.Context, tx sqlx.QueryerContext) ([]fleet.OperatingSystem, error) {
+	var os []fleet.OperatingSystem
+	if err := sqlx.SelectContext(ctx, tx, &os, `SELECT * FROM operating_systems`); err != nil {
+		return nil, err
+	}
+	return os, nil
+}
+
+func (ds *Datastore) UpdateHostOperatingSystem(ctx context.Context, hostID uint, hostOS fleet.OperatingSystem) error {
```

Comment:
Should we wrap this in a transaction (`withRetryTxx`)? Like we do in `Datastore.UpdateHostSoftware`.